examples/armv4t/gdb/host_io.rs

```diff
@@ -246,29 +242,27 @@ impl target::ext::host_io::HostIoUnlink for Emu {
 }
 
 impl target::ext::host_io::HostIoReadlink for Emu {
-    fn readlink<'a>(
-        &mut self,
-        filename: &[u8],
-        output: HostIoOutput<'a>,
-    ) -> HostIoResult<HostIoToken<'a>, Self> {
+    fn readlink<'a>(&mut self, filename: &[u8], buf: &mut [u8]) -> HostIoResult<usize, Self> {
```

Comment:
ahh shoot. based on our discussion, this current implementation is incorrect.

naively using `copy_range_to_buf` is incorrect, as it will silently truncate the response without reporting `ENAMETOOLONG`.

Instead, I think you need to do something like:

```rust
let n = copy_range_to_buf(src, ...);
if n != src.len() {
    return Err(HostIoError::Errno(HostIoErrno::ENAMETOOLONG))
}
```

if you don't like how verbose this is, you could wrap this in a `host_io.rs` local helper function.